index.js

```diff
@@ -406,19 +411,19 @@ function windowCreateFinish(json) {
     elements[json.targetID].loadURL(json.url, (typeof json.windowOptions.load !== "undefined" ? json.windowOptions.load :  {}));
     elements[json.targetID].on('blur', () => { client.write(json.targetID, consts.eventNames.windowEventBlur) })
     elements[json.targetID].on('close', (e) => {
-        if (typeof json.windowOptions.custom !== "undefined") {
-            if (typeof json.windowOptions.custom.messageBoxOnClose !== "undefined") {
-                let buttonId = dialog.showMessageBoxSync(null, json.windowOptions.custom.messageBoxOnClose)
-                if (typeof json.windowOptions.custom.messageBoxOnClose.confirmId !== "undefined" && json.windowOptions.custom.messageBoxOnClose.confirmId !== buttonId) {
+        if (typeof windowOptions[json.targetID].custom !== "undefined") {
```

Comment:
I think you should add `typeof windowOptions[json.targetID] !== "undefined"` as well in the `if`:

`if (typeof windowOptions[json.targetID] !== "undefined" && typeof windowOptions[json.targetID].custom !== "undefined")`